### DISCUSSION TITLE
Hardware Model and ECID of cloned image

### DISCUSSION THREAD

**Discussion by gerarparra0** [cirruslabs/tart#404] [General]
Is it possible to modify the hardwareModel and ecid of a cloned image?
I was wondering if it was possible to modify this information since the Virtualization Framework allows for creating new NVRAMs without providing any other attribute.

**Comment by fkorotkov** [comment]
My understanding was that this information is used upon macOS installation process and you'll replace it, than the VM won't boot. Note that I haven't tried swapping disks of VMs and it's my theoretical assumption. What is your use case for trying to modify these properties?

**Reply by gerarparra0** [reply]
I am trying to find a way to speed up the process of creating new instances with different SN/HardwareID to use together with a packer script and execute some commands on it. This process requires the SN to be different, though and this is what I'm trying to research. I have done some testing by generating new NVRAMs and modifying the config.json file tart creates, but as you mentioned, the image does not boot (tart accepts the base64 values though!). Are you aware of any other method that could be used instead of this? Thanks for your quick response btw!

**Reply by gerarparra0** [reply]
Oh and, the instances have to be running simultaneously. Sorry I left that part out.

**Reply by fkorotkov** [reply]
Oh. I guess it a duplicate #59. Does it look like it will solve your issue?

**Reply by gerarparra0** [reply]
Thanks for that! I will git it a try and report back! 🙏

**Reply by gerarparra0** [reply]
That did not work, unfortunately. Tart accepts the NVRAM, but the Virt. Framework hangs when booting. We will have to accept the fact that it's not currently possible. Thank you for the help!
